### PR TITLE
feat: create page List Cinema, Create Cinema, Detail Cinema

### PR DESCRIPTION
**Pull request by cuongdc03**
create page List Cinema, Create Cinema, Detail Cinema
feat/86epuet71

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -48,7 +50,24 @@ function App() {
             </>
           }
         />
-
+        <Route
+          path='/cinema/create'
+          element={
+            <>
+              <PageTitle title='Create Cinema | CineStar' />
```
Comment (ces-anhlam) [inline src/App.tsx:57]: create HOC function to add `PageTitle` and constant for the page title

File: src/components/Cinema/Cinema.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+const Cinema: React.FC = () => {
+  const [cinemas, setCinemas] = useState<cinema[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState<string | null>(null)
+
+  const handleDeleteCinema = async (cinema: cinema) => {
+    try {
+      await deleteCinema(cinema.id, setCinemas)
+    } catch (error) {}
```
Comment (ces-anhlam) [inline src/components/Cinema/Cinema.tsx:15]: throw error??

File: src/components/Cinema/Cinema.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from 'react'
+import { cinema } from '../../types/cinema'
+import Breadcrumb from '../Breadcrumbs/Breadcrumb'
+import { deleteCinema, getCinemas, updateCinema } from '../../apis/cinema'
+import Table from '../Table'
+
+const Cinema: React.FC = () => {
+  const [cinemas, setCinemas] = useState<cinema[]>([])
+  const [isLoading, setIsLoading] = useState(true)
```
Comment (ces-anhlam) [inline src/components/Cinema/Cinema.tsx:9]: Do you use this?

File: src/components/Cinema/Cinema.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  const [error, setError] = useState<string | null>(null)
```
Comment (ces-anhlam) [inline src/components/Cinema/Cinema.tsx:10]: Please check the unused variable

File: src/components/Cinema/CreateCinema.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  } = useForm<cinema>()
+  const [provinceCities, setProvinceCities] = useState<provincecity[]>([])
+  const navigate = useNavigate()
+
+  const fetchProvinceCities = async () => {
+    try {
+      const fetchedProvinceCities = await getprovinceCities()
+      setProvinceCities(fetchedProvinceCities || [])
+    } catch (error) {
+      console.error('Error fetching province cities:', error)
```
Comment (ces-anhlam) [inline src/components/Cinema/CreateCinema.tsx:28]: remove log

File: src/components/Cinema/CreateCinema.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+      <form onSubmit={handleSubmit(onSubmit)} className='grid grid-cols-1 gap-9 sm:grid-cols-2'>
+        <div className='flex flex-col gap-9'>
+          <div className='rounded-sm border border-stroke bg-white shadow-default dark:border-strokedark dark:bg-boxdark'>
+            <div className='border-b border-stroke px-6.5 py-2 dark:border-strokedark'></div>
+            <div className='flex flex-col gap-5.5 p-6.5'>
+              <div>
+                <label className='mb-3 block font-extrabold text-black dark:text-white'>
+                  Cinema Name <span className='text-red-500'>*</span>
+                </label>
+                <input
```
Comment (ces-anhlam) [inline src/components/Cinema/CreateCinema.tsx:71]: create Input component to reuse

File: src/components/Cinema/DetailCinema.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        const response = await getCinema(Number(id))
+        setValue('id', response.id)
+        setValue('name', response.name)
+        setValue('address', response.address)
+        setValue('provinceCityId', response.provinceCityId)
+        setValue('provinceCity', response.provinceCity)
+      } catch (error) {
+        console.error('Error fetching cinema details:', error)
+      } finally {
+        setLoading(false)
```
Comment (ces-anhlam) [inline src/components/Cinema/DetailCinema.tsx:39]: setLoading after `fetchProvinceCities`

File: src/components/Cinema/DetailCinema.tsx [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+          id: city.id,
+          name: city.name
+        }))
+        setProvinceCities(data)
+      } catch (error) {
+        console.error('Error fetching province cities:', error)
+      }
+    }
+
+    fetchCinemaDetail()
```
Comment (ces-anhlam) [inline src/components/Cinema/DetailCinema.tsx:56]: await??

File: src/components/Cinema/DetailCinema.tsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+    watch,
+    formState: { errors }
+  } = useForm<cinema>()
+  const selectedProvinceCityId = watch('provinceCityId')
+
+  useEffect(() => {
+    const fetchCinemaDetail = async () => {
+      try {
+        const response = await getCinema(Number(id))
+        setValue('id', response.id)
```
Comment (ces-anhlam) [inline src/components/Cinema/DetailCinema.tsx:31]: destructuring `response`

**Review by ces-khanhhoang — APPROVED** [review]

File: src/apis/provincecity.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+import { ProvinceCityType } from '../types/ProvinceCity'
+import { customFetch, showError } from '../util/http'
+import { PROVINCE_CITY_URL } from './constant'
+
+export const getprovinceCities = async (): Promise<ProvinceCityType[]> => {
```
Comment (ces-anhlam) [inline src/apis/provincecity.ts:5]: camelCase

Comment (ces-anhlam) [reply src/apis/provincecity.ts:5]: please use `Code Spell Checker` extension to check spell

File: src/components/MapComponent/MapComponent.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+      } finally {
+        setLoading(false)
+      }
+    }
+
+    fetchCoordinates()
+  }, [address, provinceCity])
+
+  return (
+    <div style={{ height: '400px', width: '100%' }}>
```
Comment (ces-anhlam) [inline src/components/MapComponent/MapComponent.tsx:41]: `tailwind`

File: src/components/MapComponent/MapComponent.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+      {loading ? (
+        <div>Loading map...</div>
+      ) : (
+        <MapContainer center={mapCenter} zoom={25} style={{ height: '100%', width: '100%', zIndex: '10' }}>
```
Comment (ces-anhlam) [inline src/components/MapComponent/MapComponent.tsx:45]: same above

File: src/components/Table.tsx [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+          <Link
+            to='create'
+            className='relative mr-2 items-center justify-center rounded-md border border-primary px-10 py-2 text-center font-medium text-primary hover:bg-opacity-90'
+          >
+            Create
+          </Link>
+        </div>
+      </div>
+      <ThemeProvider theme={theme}>
+        <div style={{ height: 600, width: '100%' }}>
```
Comment (ces-anhlam) [inline src/components/Table.tsx:160]: same above

**Review by ces-phatnguyen — APPROVED** [review]

**Pull request merged**